tensorflow_lite_support/c/test/task/vision/image_classifier_test.cc

```diff
@@ -0,0 +1,302 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow_lite_support/c/task/vision/image_classifier.h"
+
+#include <string.h>
+
+#include "tensorflow_lite_support/c/task/processor/classification_result.h"
+#include "tensorflow_lite_support/c/task/vision/core/frame_buffer.h"
+#include "tensorflow_lite_support/c/test/task/vision/utils/image_utils.h"
+#include "tensorflow_lite_support/cc/port/gmock.h"
+#include "tensorflow_lite_support/cc/port/gtest.h"
+#include "tensorflow_lite_support/cc/test/test_utils.h"
+
+namespace tflite {
+namespace task {
+namespace vision {
+namespace {
+
+using ::tflite::task::JoinPath;
+
+constexpr char kTestDataDirectory[] =
+    "tensorflow_lite_support/cc/test/testdata/task/vision/";
+// Quantized model.
+constexpr char kMobileNetQuantizedWithMetadata[] =
+    "mobilenet_v1_0.25_224_quant.tflite";
+
+CImageData LoadImage(const char* image_name) {
+  return CDecodeImageFromFile(
+      JoinPath("./" /*test src dir*/, kTestDataDirectory, image_name).data());
+}
+
+TEST(CImageClassifierFromOptionsTest, FailsWithMissingModelPath) {
+  TfLiteImageClassifierOptions options = {0};
+  TfLiteImageClassifier* image_classifier =
+      TfLiteImageClassifierFromOptions(&options);
+  ASSERT_EQ(image_classifier, nullptr);
```

Comment:
Use EXPECT. ASSERT_EQ will make the test fail. 

Have you tried running this test? i.e. `bazel test tensorflow_lite_support/c/test/task/vision:image_classifier_test`?

Reply:
Yes. It works fine. If ASSERT_EQ fails, it aborts only the current test. If it succeeds, it continues with the rest of the code in the function. Since there was no cleanup code after the ASSERT, didn't make sense to me to use EXPECT_EQ. Since this is a fails case for missing model path, ideally the image_classifier would never be non null.  Hence didn't add any cleanup. But logically, we should use EXPECT_EQ and delete the image_classifier if it is non null after. Let me know if this should be done.

Reply:
ASSERT works in this case. But theoretically, you want to use EXPECT to validate an invariant for your testing purposes, while using ASEERT to check invariants that are critical to execute the rest of your test. In this case, image_classifier is a test invariant that you want to check, so use EXPECT makes more sense.  Once you implemented error_status, you may also want to verify the error message and error code in this test. Therefore using ASSERT is inappropriate.

Reply:
Got it. I have fixed this. But haven't added the cleanup for this test in the update since the structure of test will change to verifying error codes rather than nullptr once it's added. Updated tests run fine from my end.